### ISSUE TITLE
Divide examples in three sections

### ISSUE THREAD

**Issue by LilianBoulard** [skrub-data/skrub#690]
### Describe the issue linked to the documentation

In the likes of [MNE-Python](https://mne.tools/stable/auto_examples/index.html), I think we should divide our examples into subsections, with our current scope, probably "joining", "encoding" and "cleaning".

### Suggest a potential alternative/fix

For reference, this is something that was proposed in #546.

**Comment by rcap107** [comment]
I don't think this is needed, but worth keeping in mind if we rework the gallery

**Comment by rcap107** [comment]
closing this issue as we are considering a much more radical reworking of the documentation

**Comment by rcap107** [comment]
Reopening after IRL discussions. We still want to do this after all.

**Comment by rcap107** [comment]
Currently, the examples are grouped in "Examples" (so no grouping) and "Skrub DataOps" (which contain only Data Ops).

We might want to group the examples in sections like:
- Getting started -- which would contain only a single example, or none depending on how we rework the user guide, or maybe we could move ApplyToCols/TableVectorizer here? 
- Encoders -- which contain the table vectorizer, string encoder, datetime encoder (which is a problem on its own), squashing scaler? 
- Data Ops
- Joiners and utils (deduplicate)

The order is intentional to reduce the emphasis on joiners. 

Functionally, this is very easy to do: grouping examples can be done by creating a folder for each section in `examples/`, and moving the relative examples inside.

**Comment by maureen-githaiga** [comment]
i'll work on this.
